File: 66-plus-one/66-plus-one.js
/**
 * @param {number[]} digits
 * @return {number[]}
 */
var plusOne = function (digits = []) {
  let r = 0;
  let result = [];
  for (var i = digits.length - 1; i >= 0; i--) {
    let item = digits[i];
    if (i == digits.length - 1) {
      item += 1 + r;
    } else item += r;
    r = 0;
    if (item >= 10) {
      item -= 10;
      r = 1;
    }
    result.push(item);
    if (i == 0 && r == 1) {
      result.push(r)
    }
  }
  return result.reverse();
}

File: 66-plus-one/66-plus-one.go
func firstUniqChar(s string) int {
	fmt.Println('c' - 'a')
	t := make([]int, 26)
	fmt.Println(t)
	for _, v := range s {
		t[v-'a']++
	}
	for i, v := range s {
		if t[v-'a'] == 1 {
			return i
		}
	}
	return -1
}

func plusOne(digits []int) []int {
	r := 0
	result := []int{}
	for i := len(digits) - 1; i >= 0; i-- {
		item := digits[i]
		if i == len(digits)-1 {
			item = item + r + 1
		} else {
			item = item + r
		}

		r = 0
		if item >= 10 {
			item = item - 10
			r = 1
		}

		result = append(result, item)
		if i == 0 && r != 0 {
			result = append(result, r)
		}
	}

	for i, j := 0, len(result)-1; i < j; i, j = i+1, j-1 {
		result[i], result[j] = result[j], result[i]
	}

	return result
}